<launch>

    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="x_pos" default="0.0"/>
    <arg name="y_pos" default="0.0"/>
    <arg name="z_pos" default="0.0"/>



    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <!-- arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/empty.world"/ -->
        <arg name="world_name" value="$(find ptdr)/worlds/search_and_rescue_world.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <param name="/use_sim_time" value="true" /> 
    <!-- node name="gazebo" pkg="gazebo" type="gazebo" args="-u $(find ptdr)/worlds/empty.world" respawn="false" output="screen"/ -->

    <group ns="robot0">

        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
        <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf0" args="-urdf -model turtlebot30_$(arg model) -x 0 -y 0 -z $(arg z_pos) -param robot_description" />

    </group>

    <group ns="robot1">

        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
        <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf1" args="-urdf -model turtlebot31_$(arg model) -x 1 -y 1 -z $(arg z_pos) -param robot_description" />

    </group>

    <group ns="robot2">

        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
        <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf2" args="-urdf -model turtlebot32_$(arg model) -x -1 -y -1 -z $(arg z_pos) -param robot_description" />

    </group>

</launch>